go/engine/team_post.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package engine
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/keybase1"
+
+	"golang.org/x/crypto/nacl/box"
+)
+
+type NewTeamEngine struct {
+	libkb.Contextified
+	name string
+}
+
+func NewNewTeamEngine(g *libkb.GlobalContext, name string) *NewTeamEngine {
+	return &NewTeamEngine{
+		Contextified: libkb.NewContextified(g),
+		name:         name,
+	}
+}
+
+func (e *NewTeamEngine) Name() string {
+	return "NewTeam"
+}
+
+func (e *NewTeamEngine) Prereqs() Prereqs {
+	return Prereqs{
+		Device: true,
+	}
+}
+
+func (e *NewTeamEngine) RequiredUIs() []libkb.UIKind {
+	return []libkb.UIKind{
+		libkb.LogUIKind,
+		libkb.SecretUIKind,
+	}
+}
+
+func (e *NewTeamEngine) SubConsumers() []libkb.UIConsumer {
+	return []libkb.UIConsumer{}
+}
+
+func (e *NewTeamEngine) Run(ctx *Context) (err error) {
+	defer e.G().Trace("NewTeamEngine", func() error { return err })()
+
+	me, err := libkb.LoadMe(libkb.NewLoadUserArg(e.G()))
+	if err != nil {
+		return err
+	}
+
+	ska := libkb.SecretKeyArg{
+		Me:      me,
+		KeyType: libkb.DeviceSigningKeyType,
+	}
+	sigKey, err := e.G().Keyrings.GetSecretKeyWithPrompt(ctx.SecretKeyPromptArg(ska, "to create a new team"))
+	if err != nil {
+		return err
+	}
+	if err = sigKey.CheckSecretKey(); err != nil {
+		return err
+	}
+
+	// TODO BEFORE LANDING: Get the owner's eldest seqno and format it as part of the name here, to handle resets.
+	ownerName := me.GetName()
+
+	id := libkb.RootTeamIDFromName(e.name)
+	teamSection := libkb.TeamSection{
+		Name: e.name,
+		ID:   id,
+	}
+	teamSection.Members.Owner = []string{ownerName}
+	teamSection.Members.Admin = []string{}
+	teamSection.Members.Writer = []string{}
+	teamSection.Members.Reader = []string{}
+	teamSection.SharedKey.Boxes = map[string]string{}
```

Comment:
we'll need this stuff elsewhere, almost certainly.  maybe do a refactor now? otherwise we might be getting merge conflicts down the road...